src/auth/src/credentials/user_credential.rs

```diff
@@ -81,15 +81,12 @@ impl TokenProvider for UserTokenProvider {
         let resp = builder
             .send()
             .await
-            .map_err(|e| CredentialError::new(false, e.into()))?;
+            .map_err(CredentialError::non_retryable)?;
```

Comment:
why is this error not retryable?

if it is a connection error or something like that, it should be possible to retry?

Reply:
Made retryable in #700